pkg/grillelasticsearch/assertions.go

```diff
@@ -0,0 +1,53 @@
+package grillelasticsearch
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Swiggy/grill"
+	"io"
+	"reflect"
+)
+
+func (ge *ElasticSearch) AssertItem(index, docId string, expected json.RawMessage) grill.Assertion {
+	return grill.AssertionFunc(func() error {
+		output, err := ge.elasticSearch.Client.Get(index, docId)
+		if err != nil {
+			return err
+		}
+
+		if output.StatusCode != itemFoundStatusCode {
+			return fmt.Errorf("invalid status code, got=%v, want=%v ", output.StatusCode, itemFoundStatusCode)
+		}
+
+		bytes, err := io.ReadAll(output.Body)
+		if err != nil {
+			return err
+		}
+
+		var item getItem
+		err = json.Unmarshal(bytes, &item)
+		if err != nil {
+			return err
+		}
+
+		if !reflect.DeepEqual(item.Source, expected) {
+			return fmt.Errorf("invalid item, got=%v, want=%v", item.Source, expected)
+		}
+		return nil
+	})
+}
+
+func (ge *ElasticSearch) AssertNoItem(index, docId string) grill.Assertion {
```

Comment:
this method is not required. its !AssertItem()

Reply:
This asserts that item doesn't exist, whereas AssertItem's main purpose is to assert on items's deep equality. AssertNoItem was created with the intention to assert for delete operation. 

Eg, if the user deletes an item and asserts using !AssertItem returns true, its not completely sure that the item is deleted or not. Even if the item was modified, !AssertItem will return true [error]

This was inspired from stretchr [NoError](https://github.com/stretchr/testify/blob/af4d8a66cf17a72a6273211ee1cc7b47930abc4a/assert/assertions.go#L1490)

Reply:
have 1 method -> AssertItemCount similar to other helpers. 
inputs are item and the expected count. 

for not exists count can be passed as 0.

Reply:
Not completely aligned on having a single method, AssertItemCount to assert on two things. I see two issues
1. From design perspective, its not a pure method. It will be doing 2 things, asserting on count and asserting on item
2. From use case perspective, if someone wants to assert just on count, they will also have to pass items for assertion

IMO, the following assertions should cover majority of use cases
1. AssertItem
2. AssertCount
3. AssertAll
4. AssertNoItem

For our particular use case, right now, we just need AssertItem. For now, we can go ahead with just AssertItem, and other methods incrementally.

Open to discussion.

Reply:
assertitem ,assertCount, assertnoitem are all the same thing .

asserItem -> AssertCount(item, 1)
assertCount -> AssertCount(item, count)
assertNoItem -> AssertCount(item, 0)

Reply:
Done